File: vimana/v12_c_cons/fib3.rb
#
# ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]
#
# ruby fib3.rb  2.21s user 0.10s system 95% cpu 2.419 total
#

def fib(n)
  if n < 2 then
    n
  else
    fib(n - 1) + fib(n - 2)
  end
end

puts fib(37)

File: vimana/v12_c_cons/fib3.php
<?php
/*
PHP 8.1.5 (cli) (built: Apr 16 2022 00:24:39) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.5, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.5, Copyright (c), by Zend Technologies

php fib3.php  2.00s user 0.03s system 99% cpu 2.042 total
*/

function fib($n)
{
  if ($n < 2) return $n;
  return fib($n - 1) + fib($n - 2);
}

echo fib(37) . "\n";
